testdata/project-v4/api/v1/admiral_webhook.go

```diff
@@ -29,18 +34,35 @@ var admirallog = logf.Log.WithName("admiral-resource")
 func (r *Admiral) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithDefaulter(&AdmiralCustomDefaulter{}).
 		Complete()
 }
 
 // TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 
 // +kubebuilder:webhook:path=/mutate-crew-testproject-org-v1-admiral,mutating=true,failurePolicy=fail,sideEffects=None,groups=crew.testproject.org,resources=admirales,verbs=create;update,versions=v1,name=madmiral.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &Admiral{}
+type AdmiralCustomDefaulter struct{}
+
+var _ webhook.CustomDefaulter = &AdmiralCustomDefaulter{}
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type
-func (r *Admiral) Default() {
-	admirallog.Info("default", "name", r.Name)
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type Admiral
+func (d *AdmiralCustomDefaulter) Default(ctx context.Context, obj runtime.Object) error {
+	admirallog.Info("CustomDefaulter for Admiral")
+	req, err := admission.RequestFromContext(ctx)
+	if err != nil {
+		return fmt.Errorf("expected admission.Request in ctx: %w", err)
+	}
+	if req.Kind.Kind != "Admiral" {
```

Comment:
This kind check seems pretty redundant. Shouldn't it be enough if the type case in l.59 works?

Reply:
I think we can remove this one 👍